engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeAsyncCommandsTest.scala

```diff
@@ -580,6 +612,20 @@ class RuntimeAsyncCommandsTest
     context.send(
       Api.Request(requestId, Api.PushContextRequest(contextId, item1))
     )
+    var isProgramStarted  = false
+    var iteration         = 0
+    var out: List[String] = Nil
+    while (!isProgramStarted && iteration < 100) {
+      context.out.monitor.synchronized {
+        context.out.monitor.wait(100)
```

Comment:
Ineffective. There is no point to `wait(100)` if the _program may already be started_:
- first of call check if `isProgramStated`
- if so, break the loop
- if not, `wait`
- all the looping shall be done inside of a `synchronized` block